netshare/generate/denormalize_fields.py

```diff
@@ -32,14 +41,66 @@ def _denormalize_by_fields_list(
     """
     denormalized_data = []
     dim = 0
+    word2vec_config = get_config(
+        ["pre_post_processor.config", "learn.word2vec"], default_value={}
+    )
+    if not word2vec_config:
+        logger.debug("No learn.word2vec config found, skipping the embedding model")
+
+    session_key_fields = word2vec_config.get(
+        "session_key", word2vec_config.get("metadata", [])
+    )
+    word2vec_cols = [
+        m
+        for m in (session_key_fields + word2vec_config.get("timeseries", []))
+        if "word2vec" in m.get("encoding", "")
+    ]
+    word2vec_cols_names = [
+        m["column"]
+        for m in (session_key_fields + word2vec_config.get("timeseries", []))
+        if "word2vec" in m.get("encoding", "")
+    ]
```

Comment:
in all this lines you're reading and parsing the learning configuration. It smells that the parsing logic happens twice.
Here's a suggestion that will make our life much much easier:
Add the member `is_word2vec_encoding` to `Field` with the default value `False`. 
Populate it during the instance creation in the module `setup_cross_chunks_data` line 163. 
then, here, you can just filter the given `fields_list` using this member.
I.e. this is better because of two big reasons:
1. We encapsulate the configuration parsing logic in a single place
2. you replace 16 lines of complex code with 3 lines of easy code :-)

Reply:
Another benefit will be that the rest of your logic will use the `Field` object, and not `Any`, such as this function signature: `get_word2vec_type_col(word2vec_cols: List[Any])`

Reply:
Yes, I agree that adding word2vec flag into field is a great idea. Previously, I also had a chat with @sniperyyc on this stuff. I'm not doing that because the modification may include a lot of files. I'm bit afraid to have a large conflit with you on the branch `code-refactor` if you have some continue update on that. Since the work on `code-refactor` is almost done, I can do that!

Reply:
I don't have any work left in `code-refactor`, so don't worry about more conflicts :)

Reply:
> 

Yes, I remembered our discussion. Adding a `word2vec` field is definitely a more generic/long-term solution to this.

Reply:
> Yes, I agree that adding word2vec flag into field is a great idea. Previously, I also had a chat with @sniperyyc on this stuff. I'm not doing that because the modification may include a lot of files. I'm bit afraid to have a large conflit with you on the branch `code-refactor` if you have some continue update on that. Since the work on `code-refactor` is almost done, I can do that!

Yes, I remembered our discussion. Adding a word2vec field is definitely a more generic/long-term solution to this.